sweagent/agent/models.py

```diff
@@ -227,7 +227,9 @@ def __init__(self, args: ModelArguments, commands: list[Command]):
             self.api_model = cfg["AZURE_OPENAI_DEPLOYMENT"]
             self.client = AzureOpenAI(api_key=cfg["AZURE_OPENAI_API_KEY"], azure_endpoint=cfg["AZURE_OPENAI_ENDPOINT"], api_version=cfg.get("AZURE_OPENAI_API_VERSION", "2024-02-01"))
         else:
-            self.client = OpenAI(api_key=cfg["OPENAI_API_KEY"])
+            self.api_base_url = cfg.get("OPENAI_BASE_URL", None)
+            self.api_model = cfg.get("OPENAI_MODEL_NAME", None)
```

Comment:
This overrides the setting of the model name via `self.args.model_name`. It's actually also incorrectly done with the `AZURE_OPENAI_DEPLOYMENT`. I will separately clean this up. `self._api_model` is set in `BaseModel.__init__`

Reply:
I did go back in forth myself with using the AZURE_OPENAI_DEPLOYMENT but in the end decided since it was labeled azure openai it was specific to azure along with its endpoint setting, so a separate one was created for the plain openai that goes along with the normal openai api key setting.

as far as overriding the model name that was the point so if you set the settings to a local llm 
OPENAI_BASE_URL: 'http://localhost:1234/v1'
OPENAI_MODEL_NAME: 'TheBloke/stablelm-zephyr-3b-GGUF/stablelm-zephyr-3b.Q4_K_S.gguf'



LM Studio Server Console Example
```
# Load a model, start the server, and run this example in your terminal
# Choose between streaming and non-streaming mode by setting the "stream" field
curl http://localhost:1234/v1/chat/completions \
-H "Content-Type: application/json" \
-d '{ 
  "model": "TheBloke/stablelm-zephyr-3b-GGUF/stablelm-zephyr-3b.Q4_K_S.gguf",
  "messages": [ 
    { "role": "system", "content": "Always answer in rhymes." },
    { "role": "user", "content": "Introduce yourself." }
  ], 
  "temperature": 0.7, 
  "max_tokens": -1,
  "stream": true
}'
```

Reply:
No, I mean that the model name should be specified as a command line option (`--model_name`), not in `keys.cfg`.

Reply:
o yeah I see I was not looking at the command line argument also